src/torchcodec/decoders/_core/VideoDecoder.h

```diff
@@ -424,6 +437,11 @@ class VideoDecoder {
   // STREAM AND METADATA APIS
   // --------------------------------------------------------------------------
 
+  void addStream(
+      int streamIndex,
+      AVMediaType mediaType,
+      const VideoStreamOptions& videoStreamOptions = VideoStreamOptions());
+
```

Comment:
We should probably make `addStream()` a private member function, and force callers to use the named ones. (Apologies if that is already the case; I have difficulty tracking where the public and private modifiers are in the PR.)